torch/csrc/autograd/input_metadata.h

```diff
@@ -67,9 +104,25 @@ struct InputMetadata {
     return at::zeros(shape_, options_);
   }
 
+  at::IntArrayRef compute_shape(const at::Tensor& t){
+    if(t.is_nested()){
+      return at::IntArrayRef{};
+    }
+    return t.sizes();
+  }
+
+  c10::optional<const at::Tensor> compute_nested_shape_from_tensor(const at::Tensor& t){
+    if(t.is_nested()){
+      auto nested_impl = at::native::get_nested_tensor_impl(t);
+      return nested_impl->get_nested_size_tensor();
+    }
+    return c10::optional<const at::Tensor> {};
+  }
+
 private:
   const at::TensorOptions options_;
   at::DimVector shape_;
+  c10::optional<const at::Tensor> nested_shape_;
```

Comment:
I don't think this is the right way to go.
You will have similar issues for all the formulas that save sizes. And we don't want to add these attributes for all of them.
We need to find a way to store the information you need into shape I think.